addons/mail/static/src/messaging/entity/message/message.js

```diff
@@ -98,7 +98,10 @@ function MessageFactory({ Entity }) {
             if ('author_id' in data) {
                 if (!data.author_id) {
                     data2.author = [['unlink-all']];
-                } else {
+                } else if (data.author_id[0] !== 0) {
+                    // partner id 0 is a hack of message_format to refer to an
+                    // author non-related to a partner. display_name equals
+                    // email_from, so this is omitted due to being redundant.
```

Comment:
What is redundant in that case?

Reply:
`data.author_id[1] === data.email_from`, so it's useless to do something with `data.author_id[1]`. Comment already say that `data.author_id[0] === 0` is a hack that shouldn't be meant as a partner id at all, so basically:

- `data.author_id[0] === 0` => hack (not a partner id)
- this implies `data.author_id[1] === data.email_from` (`data.author_id[1]` is therefore redundant).

All this to say to summarize that `author_id[0] === 0` should be ignored in conversion of data.